### DISCUSSION TITLE
Legends for View System?

### DISCUSSION THREAD

**Discussion by YohannesTz** [patrykandpatrick/vico#416] [Questions] [answered]
I am not sure if this has been asked before but I don't find any samples for legends in the Views system. I am using Java with XML if it matters.

**Comment by Gowsky** [comment, marked as answer]
Hello @YohannesTz, [`Chart7`](https://github.com/patrykandpatrick/vico/blob/28f3bb27aadb4c6fd341e020974b8cb24282111b/sample/src/main/java/com/patrykandpatrick/vico/sample/showcase/charts/Chart7.kt#L82) in the sample app uses a vertical legend. It uses a [Composable function](https://github.com/patrykandpatrick/vico/blob/28f3bb27aadb4c6fd341e020974b8cb24282111b/sample/src/main/java/com/patrykandpatrick/vico/sample/showcase/charts/Chart7.kt#L109) to create an instance of `VerticalLegend`, however, it would be roughly the same for the view system. Please keep in mind that even though Vico may work with Java, it was not written with Java in mind. You won’t be able to access e.g. default constructor or function arguments.

**Reply by YohannesTz** [reply]
thanks for the quick reply. I was experimenting with one of those, and it seems like I can't initialize the TextComponent inside [LegendItem](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.legend/-legend-item) . how do you suggest I go around this? or should I give up?

**Reply by Gowsky** [reply]
You can try with the [`TextComponentKt.textComponent`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.component.text/text-component) function or [`TextComponent.Builder`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.component.text/-text-component/-builder/) (which is not a standard Java builder class).

**Reply by YohannesTz** [reply]
@Gowsky thank you so much.

**Comment by YohannesTz** [comment]
for anyone in the future this is how I did mine. (very basic)

Edit: converted to build more legends.
```java
    public Legend getLegends(String... names) {
        TextComponent textComponent = new TextComponent.Builder()
                .build();
        ShapeComponent shapeComponent = new ShapeComponent();
        List<LegendItem> legendItems = new ArrayList<>();

        for (String name: names) {
            legendItems.add(new LegendItem(shapeComponent, textComponent, name));
        }

        return new HorizontalLegend(
            legendItems, 10f,10f,10f, 10f, new MutableDimensions(10f, 10f)
        );
    }
```

**Reply by patrickmichalik** [reply]
Thanks for sharing!
